test/testdrive/kafka-headers.td

```diff
@@ -0,0 +1,176 @@
+{"key": "fish"} {"f1": "fishval", "f2": 1000}
+
+> SELECT key, f1, f2, headers[1].value as gus1, headers[2].value as gus2 from headers_src
+key     f1       f2     gus1     gus2
+-------------------------------------------
+fish    fishval  1000   a        b
+fish2   fishval  1000   <null>   <null>
+
+
+# Works with other includes
+$ kafka-create-topic topic=headers_also
+
+$ kafka-ingest format=avro topic=headers_also key-format=avro key-schema=${keyschema} schema=${schema} headers={"gus":"gusfive"}
+{"key": "fish"} {"f1": "fishval", "f2": 1000}
+
+> CREATE MATERIALIZED SOURCE headers_also
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC
+  'testdrive-headers_also-${testdrive.seed}'
+  KEY FORMAT AVRO USING SCHEMA '${keyschema}'
+  VALUE FORMAT AVRO USING SCHEMA '${schema}'
+  INCLUDE HEADERS, PARTITION
+  ENVELOPE UPSERT
+
+> SELECT key, f1, f2, list_length(headers), partition from headers_also
+key     f1       f2     list_length    partition
+-----------------------------------------------
+fish    fishval  1000   1             0
+
+# esoteric ingestions
+$ kafka-ingest format=avro topic=headers_also key-format=avro key-schema=${keyschema} schema=${schema} headers={"gus": "gus=five"}
+{"key": "fish"} {"f1": "fishval", "f2": 1000}
+
+> SELECT key, f1, f2, headers[1].value as gus, partition from headers_also
+key     f1       f2     gus           partition
+-----------------------------------------------
+fish    fishval  1000   gus=five      0
+
+# null header
+$ kafka-ingest format=avro topic=headers_also key-format=avro key-schema=${keyschema} schema=${schema} headers={"gus": null}
+{"key": "fish"} {"f1": "fishval", "f2": 1000}
+
+> SELECT key, f1, f2, headers[1].value as gus, partition from headers_also
+key     f1       f2     gus           partition
+-----------------------------------------------
+fish    fishval  1000   <null>        0
+
+# conflicting naming
+$ set schemaheaders={
+        "type" : "record",
+        "name" : "test",
+        "fields" : [
+            {"name":"headers", "type":"string"}
+        ]
+    }
+
+$ kafka-create-topic topic=headers_conflict
+
+> CREATE MATERIALIZED SOURCE headers_conflict
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC
+  'testdrive-headers_conflict-${testdrive.seed}'
+  KEY FORMAT AVRO USING SCHEMA '${keyschema}'
+  VALUE FORMAT AVRO USING SCHEMA '${schemaheaders}'
+  INCLUDE HEADERS
+  ENVELOPE UPSERT
+
+# conflict!!
+! SELECT headers from headers_conflict
+contains:column reference "headers" is ambiguous
+
+# both columns exist
+> SHOW COLUMNS from headers_conflict
+name       nullable  type
+-------------------------
+headers    false     list
+headers    false     text
+key        false     text
+
+# No meaningful way to get data out in td because its a list
+# TODO(guswynn): actually check against this error message
+# when https://github.com/MaterializeInc/materialize/pull/11003
+# lands
+# ! SELECT * from headers_conflict
+# contains:???
```

Comment:
Mentioned above, but you can use `headers::text` in the query to be able to see the values here!

Reply:
hmm, I cant because there are 2 `headers` columns, should I just ban this behavior from being possible at the create-source layer?

Reply:
Ahh, yeah, we should do that. We did it for views, and it was helpful. Let's do that in a separate change, though. I filed #11637 to track.